multimodal/src/autogluon/multimodal/models/ner.py

```diff
@@ -0,0 +1,265 @@
+    def text_segment_ids_key(self):
+        return f"{self.prefix}_{TEXT_SEGMENT_IDS}"
+
+    @property
+    def text_valid_length_key(self):
+        return f"{self.prefix}_{TEXT_VALID_LENGTH}"
+
+    @property
+    def text_token_word_mapping_key(self):
+        return f"{self.prefix}_{TOKEN_WORD_MAPPING}"
+
+    @property
+    def text_word_offsets_key(self):
+        return f"{self.prefix}_{WORD_OFFSETS}"
+
+    @property
+    def text_column_prefix(self):
+        return f"{self.text_token_ids_key}_{COLUMN}"
+
+    @property
+    def text_feature_dim(self):
+        return self.model.config.hidden_size
+
+    @property
+    def label_key(self):
+        return f"{self.prefix}_{LABEL}"
+
+    def forward(
+        self,
+        batch: dict,
+    ):
+        """
+        Parameters
+        ----------
+        batch
+            A dictionary containing the input mini-batch data.
+            We need to use the keys with the model prefix to index required data.
+
+        Returns
+        -------
+            A dictionary with logits and features.
+        """
+        text_token_ids = batch[self.text_token_ids_key]
+        if self.disable_seg_ids:
+            text_segment_ids = None
+        else:
+            text_segment_ids = batch[self.text_segment_ids_key]
+        text_valid_length = batch[self.text_valid_length_key]
+        steps = torch.arange(0, text_token_ids.shape[1]).type_as(text_valid_length)
+        text_masks = (steps.reshape((1, -1)) < text_valid_length.reshape((-1, 1))).type_as(text_token_ids)
+
+        if self.is_t5:
+            # For the T5 model, we will only use the encoder to encode the sentence. This is adopted in
+            # "Sentence-T5 (ST5): Scalable Sentence Encoders from Pre-trained Text-to-Text Models"
+            # (https://aclanthology.org/2022.findings-acl.146.pdf).
+            inputs_embeds = self.model.encoder.embed_tokens(text_token_ids)
+            if self.gradient_checkpointing:
+                # FIXME(?) This is a hack! We added a DummyLayer to ensure that the
+                #  gradient checkpointing will assign output layer as require_grad=True
+                #  Reference: https://discuss.pytorch.org/t/checkpoint-with-no-grad-requiring-inputs-problem/19117/9
+                inputs_embeds = self.dummy_layer(inputs_embeds)
+            outputs = self.model.encoder(
+                inputs_embeds=inputs_embeds,
+                attention_mask=text_masks,
+            )
+        else:
+            outputs = self.model(
+                input_ids=text_token_ids,
+                token_type_ids=text_segment_ids,
+                attention_mask=text_masks,
+            )
+
+        sequence_output = outputs.last_hidden_state
+        batch_size, max_len, feat_dim = sequence_output.shape
+        valid_output = torch.zeros(batch_size, max_len, feat_dim, dtype=torch.float32)
+
+        pooled_features = outputs.last_hidden_state[:, 0, :]
+
+        logits = self.classifier(sequence_output)
+
```

Comment:
May be we can just add a flag to HFAutoModelForTextPrediction to indicate that it will be used for entity-recognition problems? For example, HFAutoModelForTextPrediction(is_ner=True)

Reply:
If we have more than two columns or other modalities, it might be easier to deal with in a standalone script.